src/IO/WriteBufferFromFileDescriptorDiscardOnFailure.cpp

```diff
@@ -17,7 +17,10 @@ void WriteBufferFromFileDescriptorDiscardOnFailure::nextImpl()
 
         if ((-1 == res || 0 == res) && errno != EINTR)
         {
-            ProfileEvents::increment(ProfileEvents::CannotWriteToWriteBufferDiscard);
+            /// Never send this profile event to trace log because it may cause another
+            /// write into the same fd and likely will trigger the same error
```

Comment:
why fd can be the same when sending profile event?

Reply:
Because traces are always sent to one [pipe](https://github.com/ClickHouse/ClickHouse/blob/44fde45aa4aca792a15903e39465220259de0994/src/Common/TraceSender.cpp#L45).

Reply:
Ok, I thought "the same" is about the fd in this particular file a few lines above, which is very strange. May be just a comment is a bit unclear.